src/Lookup/ClosestReferencedEntityIdLookup.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace Wikibase\DataModel\Services\Lookup;
+
+use Wikibase\DataModel\Entity\EntityId;
+use Wikibase\DataModel\Entity\PropertyId;
+
+/**
+ * Service interface for getting the closest entity (out of a specified set),
+ * from a given starting entity. The starting entity, and the target entities
+ * are (potentially indirectly, via intermediate entities) linked by statements
+ * with a given property id.
+ *
+ * @since 3.10
+ *
+ * @license GPL-2.0+
+ * @author Marius Hoch
+ */
+interface ClosestReferencedEntityIdLookup {
+
+	/**
+	 * Get the closest entity (out of $toIds), from a given entity. The starting entity, and
+	 * the target entities are (potentially indirectly, via intermediate entities) linked by
+	 * statements with the given property id.
```

Comment:
I'm pretty happy with what I see here! :-) The only detail I'm missing is that it's not entirely clear what "linked" exactly means. It could be linked both ways: Either the $fromId contains a property linking to one of the $toIds (this is what we want to do here), or the other way around.

I think we can add a short sentence like "… linked by statements with the given property ID, pointing towards the target entities."

This relationship might become a little more obvious when we order the function arguments like this: `( EntityId $fromId, PropertyId $propertyId, array $toIds )`.

Please capitalize "ID".

Reply:
Changed.